sample_consensus/include/pcl/sample_consensus/impl/sac_model_circle.hpp

```diff
@@ -137,9 +142,10 @@ pcl::SampleConsensusModelCircle2D<PointT>::selectWithinDistance (
     inliers.clear ();
     return;
   }
-  int nr_p = 0;
-  inliers.resize (indices_->size ());
-  error_sqr_dists_.resize (indices_->size ());
+  inliers.clear ();
```

Comment:
GCC has a weird bug/feature where `reserve` results in bypassing the exponential growth of the container and the exact memory is granted. This causes quadratic time if the function with `reserve` is called in a loop.

While I do not envision this being called in a loop with growing `indices_.size()`, is it possible to either
* put a warning note somewhere
* not use reserve and pay for multiple allocations in the loop (just `push_back`, nothing else)
* (current) pay for zero constructing the array using `resize`

None of the 3 options are "good". I'd really like C++ STL to give a mix of resize + reserve: don't initialize, just gimme exponentially increasing memory.

Reply:
Interesting. Do you have a link to an article regarding that behaviour or something? Never heard of it before. Is this only an issue together with `clear()`?

Reply:
No, this is an issue with `reserve`. It's there in the docs: [reserve](https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/vector.tcc#L78) allocates exact memory, but doubles capacity for other cases.

Reply:
Sorry, I am still not sure what the problem is. We want that exact amount of memory, no? And `reserve` would only do something if the new desired capacity is bigger than the current capacity. So even if `indices_->size()` gets bigger (and I agree with you, I don't see how that would happen), this would work. And where does this exponential growth happen?

Reply:
On using `reserve` in a loop, the loop can take quadratic time instead of amortized linear since growth of the vector becomes quantized linear (step of one in worst case) instead of exponential.

We do want to allocate exact memory. The hypothetical quadratic time issue happens **IFF** `indices_->size ()` grows linearly in the loop. I can't comprehend a situation where this can happen. If @taketwo or @SergioRAgostinho can't think of a valid case where the bad-case happens, we can go ahead with this patch.

My position is of safety since I couldn't be certain that the hypothetical can never happen. To clarify:
* I asked if it was possible to replace `reserve` with something else based on a hypothetical. **If that never happens, this is resolved.**
* The issue isn't with `clear` but only `reserve` in case the hypothetical use happens in user-code. See implementation for GCC's weird behavior; clang is saner.